File: exercism/javascript/nth-prime/nth-prime.js
const fp = require('lodash/fp')

function* primes () {
  const known = new Set()
  let candidate = 2
  outer: while (true) {
    for (p of known) {
      if (candidate % p == 0) {
        candidate +=1
        continue outer
      }
    }
    yield candidate
    known.add(candidate)
    candidate += 1
  }
}

const nth = (n) => {
  if (n === 0) throw new Error('Prime is not possible')
  let i = 0
  for (let p of primes()) {
    i++
    if (i === n) return p
  }
}

module.exports = { nth }

File: exercism/go/nth-prime/nth_prime.go
package prime

// Nth returns the Nth prime
func Nth(n int) (int, bool) {
	if n == 0 {
		return 0, false
	}

	primes := make([]int, n)
	k, p := 0, 2

	for ; k < n; p++ {
		// if any prime divides p, it is not prime
		for i := 0; i < k; i++ {
			if p%primes[i] == 0 {
				goto next
			}
		}
		primes[k] = p
		k++
	next:
	}
	return p - 1, true
}
